src/Controls/src/Core/Element.cs

```diff
@@ -97,7 +98,110 @@ public string StyleId
 			}
 		}
 
-		internal virtual IReadOnlyList<Element> LogicalChildrenInternal => EmptyChildren;
+		// Leaving this internal for now.
+		// If users want to add/remove from this they can use
+		// AddLogicalChildren and RemoveLogicalChildren on the respective control
+		// if available.
+		//
+		// Ultimately I don't think we'll need these to be virtual but some controls (layout)
+		// are going to take a more focused effort so I'd rather just do that in a 
+		// separate PR. I don't think there's ever a scenario where a subclass needs
+		// to replace the backing store.
+		// If everyone just uses AddLogicalChildren and RemoveLogicalChildren
+		// and then overrides OnChildAdded/OnChildRemoved
+		// that should be sufficient
+		internal IReadOnlyList<Element> LogicalChildrenInternal
+		{
+			get
+			{
+				SetupChildren();
+				return _logicalChildrenReadonly;
+			}
+		}
+
+		private protected virtual IList<Element> LogicalChildrenInternalBackingStore
+		{
+			get
+			{
+				_internalChildren ??= new List<Element>();
+				return _internalChildren;
+			}
+		}
+
+		[EditorBrowsable(EditorBrowsableState.Never)]
+		[Obsolete("Do not use! This is to be removed! Just used by Hot Reload! To be replaced with IVisualTreeElement!")]
+		public ReadOnlyCollection<Element> LogicalChildren =>
+			new ReadOnlyCollection<Element>(new TemporaryWrapper(LogicalChildrenInternal));
+
+		void SetupChildren()
+		{
+			_logicalChildrenReadonly ??= new ReadOnlyCollection<Element>(LogicalChildrenInternalBackingStore);
+		}
+
+		internal void InsertLogicalChildInternal(int index, Element element)
+		{
+			if (element is null)
+			{
+				return;
+			}
+
+			SetupChildren();
+
+			LogicalChildrenInternalBackingStore.Insert(index, element);
+			OnChildAdded(element);
+		}
+
+		internal void AddLogicalChildInternal(Element element)
+		{
+			if (element is null)
+			{
+				return;
+			}
+
+			SetupChildren();
+
+			LogicalChildrenInternalBackingStore.Add(element);
+			OnChildAdded(element);
+		}
+
+		internal bool RemoveLogicalChildInternal(Element element)
+		{
+			if (element is null)
+			{
+				return false;
+			}
+
+			if (LogicalChildrenInternalBackingStore is null || !LogicalChildrenInternalBackingStore.Contains(element))
```

Comment:
A bit nitty, but index of will return -1 if not found, so we can avoid the contains and the second iteration.